src/PlatformAdapter/Native/Exec/NativeExecImpl.cpp

```diff
@@ -0,0 +1,253 @@
+#include "NativeExecImpl.hpp"
+
+#include "Utils/Utils.hpp"
+
+#undef INFO
+#define INFO(msg, ...)
+
+#undef DEBUG
+#define DEBUG(msg, ...)
+
+/*
+ * NativeExecImpl
+ */
+bool NativeExecImpl::schedule_eval(const std::string& code)
+{
+  if (m_state == DEAD)
+  {
+    FAILED("#NativeExecImpl::schedule_eval, error, dead state");
+    return false;
+  }
+
+  while (m_state != RUNNING)
```

Comment:
should add doc to explain m_state will be modified somewhere, otherwise it'll be spinning forever.